docs/aws/audit/redshiftmonitoring/rules/redshift_reserved_node_payment_failed.mdx

```diff
@@ -24,6 +24,230 @@ CBP
+
+   ```sh
+   aws cloudwatch put-metric-alarm --alarm-name "BillingAlarm" --metric-name "EstimatedCharges" --namespace "AWS/Billing" --statistic "Maximum" --period 21600 --threshold 100 --comparison-operator "GreaterThanOrEqualToThreshold" --dimensions Name=Currency,Value=USD --evaluation-periods 1 --alarm-actions arn:aws:sns:us-east-1:123456789012:MyTopic
+   ```
+
+3. **Update Payment Methods:**
+   Ensure that your payment methods are valid and up-to-date. You can update your payment methods through the AWS Management Console, but you can also use the AWS CLI to list your current payment methods and ensure they are valid.
+
+   ```sh
+   aws organizations list-payment-methods
+   ```
+
+4. **Automate Payment Verification:**
+   Use a script to periodically verify the status of your reserved nodes and ensure that there are no payment issues. This can be done using a combination of AWS CLI commands and a simple script.
+
+   ```sh
+   aws redshift describe-reserved-nodes --query 'ReservedNodes[?State==`payment-failed`]'
+   ```
+
+By following these steps, you can proactively monitor and manage your AWS billing and payment status to prevent Redshift Reserved Nodes from having a "Payment Failed" status.
+</Accordion>
+
+<Accordion title='Using Python'>
+To prevent the issue of Redshift Reserved Nodes having a status of "Payment Failed" using Python scripts, you can follow these steps:
+
+1. **Set Up AWS SDK (Boto3) and Authentication:**
+   Ensure you have the AWS SDK for Python (Boto3) installed and properly configured with your AWS credentials.
+
+   ```bash
+   pip install boto3
+   ```
+
+   ```python
+   import boto3
+
+   # Initialize a session using Amazon Redshift
+   session = boto3.Session(
+       aws_access_key_id='YOUR_ACCESS_KEY',
+       aws_secret_access_key='YOUR_SECRET_KEY',
+       region_name='YOUR_REGION'
+   )
+
+   redshift = session.client('redshift')
+   ```
+
+2. **Monitor Reserved Node Payment Status:**
+   Create a function to check the payment status of your reserved nodes. This function will query the reserved nodes and check their status.
+
+   ```python
+   def check_reserved_node_payment_status():
+       response = redshift.describe_reserved_nodes()
+       for node in response['ReservedNodes']:
+           if node['State'] == 'payment-failed':
+               print(f"Reserved Node ID {node['ReservedNodeId']} has a payment failed status.")
+               # Implement further actions like alerting or logging
+           else:
+               print(f"Reserved Node ID {node['ReservedNodeId']} is in good standing.")
+   ```
+
+3. **Automate Payment Status Checks:**
+   Schedule the script to run at regular intervals to continuously monitor the payment status of your reserved nodes. You can use a scheduling library like `schedule` or `APScheduler`.
+
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>